instructor/common.py

```diff
@@ -672,11 +672,34 @@ def western_blot_band_size(request):
         exclude=['primary', 'secondary', 'western_blot']
     )
     if request.method == "POST":
+        weights_by_type = ['wc_weight', 'nuc_weight', 'cyto_weight']
         formset = AntibodiesFormset(request.POST)
         if formset.is_valid():
-            entries = formset.save(commit=False)
-            for form in entries:
-                form.save()
+            antibodies = formset.save(commit=False)
+            strain_treatments = models.StrainTreatment.objects.filter(assignment=assignment)
+            for antibody in antibodies:
+                for index, field in enumerate(weights_by_type):
+                    weight_str = getattr(antibody, field)
+                    weights = [float(s) for s in weight_str.split(',')
+                               if is_float(s) and float(s) >= 0]
+                    if antibody.id:
+                        bands = models.WesternBlotBands.objects.filter(
+                            antibody__id=antibody.id,
+                            lysate_type=field.split('_')[0])
+                        for band in bands:
+                            #delete bands that that are not in the string anymore
+                            if band.weight not in weights:
```

Comment:
This is a floating point lookup so it may be prone to false negatives. However I can't find a number where this doesn't work so it's fine with me to leave it as is